plugins/user_mapping/src/user_claim.cpp

```diff
@@ -0,0 +1,64 @@
+#include "interface.h"
+#include "user_claim.hpp"
+
+#include <cstdlib>
+#include <cstring>
+#include <exception>
+#include <string>
+
+#include <spdlog/spdlog.h>
+
+namespace
+{
+	std::string claim_to_match;
+} // anonymous namespace
+
+auto init(const nlohmann::json& _config) -> void {
+	const auto claim{_config.find("irods_user_claim")};
+	if (claim == std::end(_config)) {
+		// handle bad case
+		spdlog::error("{}: Unable to find 'irods_user_claim' in provided config.", __func__);
+	}
+
+	claim_to_match = *claim;
+}
+
+auto match(const nlohmann::json& _params) -> std::optional<std::string> {
+	if (auto claim{_params.find(claim_to_match)}; claim != std::end(_params)) {
+		return {*claim};
+	}
+
+	return std::nullopt;
+}
+
+void user_mapper_init(const char* _args) {
+	try {
+		spdlog::debug("{}: recieved _args [{}]", __func__, _args);
+		init(nlohmann::json::parse(_args));
+	} catch (const std::exception& e) {
+		spdlog::debug("{}: {}", __func__, e.what());
+	}
```

Comment:
Is this for catching the exception thrown by dereferencing the `std::end` iterator in `init` (L18-20)? Are there any other exceptions which could be thrown that we don't want to mask?

Reply:
Using this discussion to simplify any further comments: https://github.com/irods/irods_client_http_api/pull/331#discussion_r1759506779

Reply:
Done.